.github/workflows/extract-translation-source-files.yml

```diff
@@ -105,9 +107,78 @@ jobs:
+        # * frontend-app-admin-portal
+        # * frontend-app-publisher
+        # * frontend-platform
+        # * frontend-enterprise?
+        # * frontend-app-learner-portal-enterprise
+        # * frontend-build?
+        # * frontend-app-learner-portal-programs
+        # * frontend-component-cookie-policy-banner
+        # * frontend-app-programs-dashboard
+        # studio-frontend has a different file structure
+        # repos with errors running extract_translations
+        # * frontend-template-application
+        repo: [frontend-app-payment, frontend-app-discussions, frontend-app-enterprise-public-catalog, frontend-app-learning, frontend-app-authn, frontend-app-course-authoring, frontend-app-learner-record, frontend-app-support-tools, frontend-component-header, frontend-component-footer, frontend-app-library-authoring, frontend-app-account, frontend-app-profile, frontend-app-ecommerce, frontend-app-gradebook, frontend-app-program-console]
     runs-on: ubuntu-latest
     needs: [setup-branch, python-translations]
+    steps:
+      # Clones the openedx-translations repo
+      - name: clone brian-smith-tcril/openedx-translations
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.setup-branch.outputs.branch }}
+
+      # Clones the  repository
+      - name: clone brian-smith-tcril/${{ matrix.repo }}
+        uses: actions/checkout@v3
+        with:
+          repository: brian-smith-tcril/${{ matrix.repo }}
+          path: ${{ matrix.repo }}
+
+      # Sets up node/npm
+      - name: setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      # Extracts the translation source files
+      - name: extract translation source files
+        run: |
+          cd ${{ matrix.repo }}
+          make extract_translations
+      # git adds only the translation source files, found in conf/locale/en
+      - name: git add the translation source files
+        id: add-sources
+        run: |
+          # set identity
+          git config --global user.email "translations-bot@openedx.org"
+          git config --global user.name "edx-transifex-bot"
+          # remove ${{ matrix.repo }}/.git so we don't commit a submodule
+          rm -rf ${{ matrix.repo }}/.git
+          # stage the transifex_input.json file generated by make
+          git add ${{ matrix.repo }}/src/i18n/transifex_input.json -f -v
+          # Check the git status of the translation source files
+          echo "GIT_STATUS=$(git status ${{ matrix.repo }}/src/i18n/transifex_input.json -s | wc -l)" >> $GITHUB_ENV
+      # Attempts to commit the translation source files if there is a difference
+      - name: git commit the translation source files
+        if: "${{ env.GIT_STATUS > 0 }}"
+        run: |
+          # commit the changes
+          git commit -m "chore: add extracted translation source files from ${{ matrix.repo }}"
+          # push changes to branch
+          git pull
```

Comment:
Since everything is running sequentially, is the `git pull` necessary here?

Reply:
tested [here](https://github.com/brian-smith-tcril/openedx-translations/actions/runs/3396813612/jobs/5648306115) and it seems to work without the pull, updating it now